tests/integration/update.bats

```diff
@@ -245,50 +245,79 @@ EOF
     check_systemd_value "TasksMax" 20
 }
 
-@test "update cgroup v1 cpu limits" {
-    [[ "$ROOTLESS" -ne 0 ]] && requires rootless_cgroup
-    requires cgroups_v1
+function check_cpu_quota() {
+	local quota=$1
+	local period=$2
+	local sd_quota=$3
+
+	if [ "$CGROUP_UNIFIED" = "yes" ]; then
+		if [ "$quota" = "-1" ]; then
+			quota="max"
+		fi
+		check_cgroup_value "cpu.max" "$quota $period"
+		check_systemd_value "CPUQuotaPerSecUSec" $sd_quota
+	else
+		check_cgroup_value "cpu.cfs_quota_us" $quota
+		check_cgroup_value "cpu.cfs_period_us" $period
+		# no systemd support in v1
+	fi
+}
 
-    # run a few busyboxes detached
-    runc run -d --console-socket $CONSOLE_SOCKET test_update
-    [ "$status" -eq 0 ]
+function check_cpu_shares() {
+	local shares=$1
+
+	if [ "$CGROUP_UNIFIED" = "yes" ]; then
+		local weight=$((1 + ((shares - 2) * 9999) / 262142))
+		check_cgroup_value "cpu.weight" $weight
+		check_systemd_value "CPUWeight" $weight
+	else
+		check_cgroup_value "cpu.shares" $shares
+		check_systemd_value "CPUShares" $shares
+	fi
+}
 
-    # check that initial values were properly set
-    check_cgroup_value "cpu.cfs_period_us" 1000000
-    check_cgroup_value "cpu.cfs_quota_us" 500000
-    check_systemd_value "CPUQuotaPerSecUSec" 500ms
+@test "update cgroup cpu limits" {
+	[[ "$ROOTLESS" -ne 0 ]] && requires rootless_cgroup
 
-    check_cgroup_value "cpu.shares" 100
-    check_systemd_value "CPUShares" 100
+	# run a few busyboxes detached
+	runc run -d --console-socket $CONSOLE_SOCKET test_update
+	[ "$status" -eq 0 ]
 
-    # systemd driver does not allow to update quota and period separately
-    if [ -z "$RUNC_USE_SYSTEMD" ]; then
-        # update cpu period
-        runc update test_update --cpu-period 900000
-        [ "$status" -eq 0 ]
-        check_cgroup_value "cpu.cfs_period_us" 900000
+	# check that initial values were properly set
+	check_cpu_quota 500000 1000000 "500ms"
+	check_cpu_shares 100
+
+	# updating cpu period alone is not allowed
+	runc update test_update --cpu-period 900000
+	[ "$status" -eq 1 ]
 
-        # update cpu quota
-        runc update test_update --cpu-quota 600000
+	# update cpu quota
+	runc update test_update --cpu-quota 600000
+	[ "$status" -eq 0 ]
+	check_cpu_quota 600000 1000000 "600ms"
+
+        # remove cpu quota
```

Comment:
nit: indent

Reply:
thanks for catching this -- fixed.

NOTE that (as I noted in the changelog message) I have converted this code (and the code I've added) to use tab instead of 4 spaces since I like tabs much more than 4 spaces, and I am changing lots of code here now.

Long term, I don't know what to do about it. I want to avoid commits just changing the whitespace as this will make it harder to look through git history, but I don't like the current way of using 4 tabs either.

Reply:
Is there any tool we can run to ensure consistency?

Reply:
`shfmt` solves this problem (and we had it in CI for a few months a while ago) but see #2396 which tracks re-adding it.